src/core/lv_indev.c

```diff
@@ -786,14 +786,24 @@ static void indev_proc_press(_lv_indev_proc_t * proc)
     if(proc->wait_until_release != 0) return;
 
     lv_disp_t * disp = indev_act->driver->disp;
+    lv_obj_t * act_scr = lv_disp_get_scr_act(disp);
+    lv_obj_t * prev_scr = lv_disp_get_scr_prev(disp);
+
+    bool consider_prev_screen = disp && act_scr && prev_scr && disp->draw_prev_over_act;
+    bool indev_obj_in_prev_screen = false;
     bool new_obj_searched = false;
 
     /*If there is no last object then search*/
     if(indev_obj_act == NULL) {
         indev_obj_act = lv_indev_search_obj(lv_disp_get_layer_sys(disp), &proc->types.pointer.act_point);
         if(indev_obj_act == NULL) indev_obj_act = lv_indev_search_obj(lv_disp_get_layer_top(disp),
                                                                           &proc->types.pointer.act_point);
-        if(indev_obj_act == NULL) indev_obj_act = lv_indev_search_obj(lv_disp_get_scr_act(disp),
+        if (indev_obj_act == NULL && consider_prev_screen) {
```

Comment:
I'm not sure about this but I think it's enough to search only on `scr_act`. `scr_act` is changed when the animation starts so the checks on `prev_scr` apply only during animation itself which is typically fast. 

So unless I miss something, I  for the sake of simplicity I'd check only `scr_act`

Reply:
So what I tested so far:
If you click while a screen is flying a way (and prev screen js drawn above act screen) the touch event will reach the not visible act screen which feels akward.

Reply:
In [this](https://forum.lvgl.io/t/a-bug-and-a-hack-fix/8280/5?u=kisvegabor) Forum topic I wrote about disabling input device during screen transitions. I think it's d result in the least confusion. What do you think?

Reply:
I am fine with such solution. I reverted any code from lv_indev.c.

Would be good to have a function to globally disable input devices, not just on screen transition, for the user available.

I don't see this as part of this PR.

Reply:
`lv_indev_enable()` is available to enable/disable an input device. However, I think it's not a good idea to use it here. It's because if a user uses  `lv_indev_enable()` and we also use it LVGL internally, the calls can overwrite each other.  

IMO all we need to do is adding a `return` [here](https://github.com/lvgl/lvgl/blob/master/src/core/lv_indev.c#L78) if there if `prev_scr != NULL`

Reply:
Well, that sounds right. Added.